### PR TITLE
Seed json

### PR DESCRIPTION
**Pull request by d-buckner**
@cadecameron @brkidder.

Here's my seeding work. The two main moving parts of this are:

- [Image downloading](https://github.com/coupon-app/service-image-carousel/compare/seed-json?expand=1#diff-03ad64ffcb8b98d22b0abb03322fc74d)

- [Compiling seed data into JSON that links to the images in S3](https://github.com/coupon-app/service-image-carousel/compare/seed-json?expand=1#diff-e6c2c72deb7a6c0c3fcc1cf7efa5c694)

Some of this could be cleaned up, I'd love any suggestions if you have any.

### CODE REVIEW

File: seed/compileSeedData/index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const imageNameString = fs
+  .readFileSync(path.resolve(__dirname, '..', 'retrieveImgs', 'images.txt'))
+  .toString();
+
+const imageNames = imageNameString.split('\n');
+
+const seedJson = [];
+
+imageNames.forEach((imageName) => {
+  const isThumnail = imageName.includes('thumbnail');
```
Comment (brkidder) [inline seed/compileSeedData/index.js]: isThum"b"nail? Totally understand if this intended just a little awkward to read.

File: seed/compileSeedData/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+  const productObj = {
+    productId,
+    imgId,
+    imgUrl: S3PATH + imageName,
+    thumbnailUrl: S3PATH + imageName.replace('.jpg', '-thumbnail.jpg'),
+  };
+
+  console.log(productObj);
```
Comment (brkidder) [inline seed/compileSeedData/index.js]: Is this console.log statement needed anymore?

File: seed/compileSeedData/index.js (hunk 1 above)
Comment (d-buckner) [reply seed/compileSeedData/index.js]: Ah, good eye! Will fix this.

File: seed/compileSeedData/index.js (hunk 2 above)
Comment (d-buckner) [reply seed/compileSeedData/index.js]: Good point, I'll remove this.

File: seed/retrieveImgs/libs.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    responseType: 'arraybuffer',
+    timeout: 20000,
+  })
+    .then((response) => Buffer.from(response.data, 'binary'))
+    .then((imageBuffer) => {
+      fs.writeFile(filePath, imageBuffer, (err) => {
+        if (err) {
+          console.log(err);
+        } else {
+          console.log('✔️  ', filePath);
```
Comment (brkidder) [inline seed/retrieveImgs/libs.js:42]: found another one. I like the symbol--gives a nice visual when things are working as you are developing the code.

**Review by brkidder — COMMENTED** [review]
Code looks good from what I see. Keep up the good work!

**Comment by d-buckner** [comment]
Thanks for looking it over!

File: server/index.js [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
 
 const app = express();
 const PORT = process.env.PORT || 3000;
 
 const PUBLIC_DIR = path.resolve(__dirname, '..', 'client', 'public');
 
-app.use(express.static(PUBLIC_DIR));
+app.use('/', express.static(PUBLIC_DIR));
+
+app.use('/api/:productId', (req, res) => {
```
Comment (cadecameron) [inline server/index.js:13]: Perhaps consider an api endpoint change (i.e. /api/images/:productId) down the track (though perhaps the proxy can take care of that)? 🤷‍♂️

Comment (d-buckner) [reply server/index.js:13]: I like the idea, I'll make a ticket for when I move back to the API interactions.

**Pull request merged**